### ISSUE TITLE
yas--all-parents can fail to find all parents if called before snippet directory is loaded

### ISSUE THREAD

**Issue by nverno** [joaotavora/yasnippet#1180]
Relating to ae36504

Sometimes `yas--all-parents` attempts to get a mode's parents before the
snippet directory containing the parent has been loaded by `yas-load-directory`.
When this happens, `yas--parents` doesnt yet contain entries for the parents
who havent been loaded, so `yas--all-parents` doesnt recurse on the parent,
thus failing to add all the parents to the `yas--all-parents` prop.

This sequence of events can happen, for example, if `yas-minor-mode` is called
in `prog-mode-hook` and a snippet directory is loaded in the subsequent major
mode's hook.

**Comment by monnier** [comment]
> This sequence of events can happen, for example, if `yas-minor-mode` is called
> in `prog-mode-hook` and a snippet directory is loaded in the subsequent major
> mode's hook.

Thanks.  I pushed a patch for this in `master`.
Can you confirm that it fixes it for you?

**Comment by stsquad** [comment]
Something has broken with git-commit-mode because now we get:

```
(yas--modes-to-activate)
(text-mode fundamental-mode fundamental-mode)
```

I'm not sure if it was these changes or something has changed with git-commit-mode. How should minor modes be handled?

**Comment by monnier** [comment]
> Something has broken with git-commit-mode because now we get:
>
> ```
> (yas--modes-to-activate)
> (text-mode fundamental-mode fundamental-mode)
> ```

Hmm... Could you be more specific?  The duplicate `fundamental-mode` is
a known problem (which I believe should be mostly harmless) for which
I have a fix pending.  Should `git-commit-mode` be included above?
IF so wherewho is this code run?  What did you mean by "now"
(i.e. which commit or which GNU-devel/MELPA version are you using)?
Also on which Emacs version?

I have not been able to reproduce a situation where
`yas--modes-to-activate` doesn't return something of the form
`(git-commit-mode ...)` when called without args in
a `git-commit-mode` buffer.

**Comment by stsquad** [comment]
It works now so possibly fixed with d12428082c03459361ad79399c35378e594d92da?

``
(yas--mode-to-activate) ->
  (text-mode fundamental-mode git-commit-mode) [2 times]
```

**Comment by monnier** [comment]
> It works now so possibly fixed with d12428082c03459361ad79399c35378e594d92da?

Might be.   I must admit that I don't know yet how `git-commit-mode`
finds its way there :-)
